src/utils/wrap_thread.hpp

```diff
@@ -0,0 +1,54 @@
+#ifndef WRAP_THREAD_HPP
+#define WRAP_THREAD_HPP
+
+#include <chrono>
+#include <future>
+#include <thread>
+
+/// More advanced then std::thread. Will join itself on destruction to avoid certain UBs. Has debug and stats.
+/// @owner hb
+class wrap_thread {
+	public:
+		wrap_thread() noexcept;
```

Comment:
///< this creates an empty-thread, that does nothing. It does not have to be joined. It does not start any thread.